nemo/collections/tts/models/magpietts.py

```diff
@@ -153,7 +146,29 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
 
         if self.model_type != 'decoder_pretrain_synthesizer':
             # Decoder pretrain synthesizer doesn't have transcript encoder/text embeddings
-            self.text_embedding = nn.Embedding(num_tokens, cfg.embedding_dim)
+            
+            if self.use_bpe_char_tokenizer:
+                # BPE char tokenizer
+                assert len(self.tokenizer.tokenizers) == 1, "BPE char tokenizer should only be used with one tokenizer"
+                tokenizer_name = self.tokenizer.tokenizer_names[0]
+                tokenizer = self.tokenizer.tokenizers[tokenizer_name]
+                subword_vocab = tokenizer.get_vocab()
+                # special tokens will be stored as it is in the char_vocab
+                # Each special token will only be mapped to one char id
+                special_vocab = {
+                    '<BOS>': self.bos_id,
+                    '<EOS>': self.eos_id,
+                }
+                self.cas_encoder = CharAwareSubwordEncoder(
+                    d_embed=cfg.embedding_dim,
+                    llm_tokenizer_vocab=subword_vocab,
+                    subword_padding_idx=self.tokenizer.pad,
+                    special_vocab=special_vocab
+                )
+            else:
+                # Regular text embedding
+                self.text_embedding = nn.Embedding(num_tokens, cfg.embedding_dim)
```

Comment:
Should we have different naming for these two embedding algorithms? Why not default both to self.text_embedding?

Reply:
I think we should use a different name because cas_encoder has a different forward interface than text_embedding. Also _embedding sound like a simple nn.Embedding layer, which can be confusing for someone reading the code, since there is a transformer also being used in CharAwareSubwordEncoder.